include/sound/sof/dai-intel.h

```diff
@@ -49,6 +49,9 @@
 /* bclk idle */
 #define SOF_DAI_INTEL_SSP_CLKCTRL_BCLK_IDLE_HIGH	BIT(5)
 
+/* DMIC max. four controllers for eight microphone channels */
+#define SOF_DAI_INTEL_DMIC_NUM_CTRL			4
```

Comment:
sorry for chiming in so late with this. This can be made a separate PR: should we add a check of `num_pdm_active <= SOF_DAI_INTEL_DMIC_NUM_CTRL`? I think it would be good to tighten up our topology consistency checks.